tests/functional/c/consider/consider_using_dict_items.py

```diff
@@ -1,5 +1,5 @@
 """Emit a message for iteration through dict keys and subscripting dict with key."""
-# pylint: disable=line-too-long,missing-docstring,unsubscriptable-object,too-few-public-methods,redefined-outer-name,use-dict-literal
+# pylint: disable=locally-disabled,useless-suppression,suppressed-message,line-too-long,missing-docstring,unsubscriptable-object,too-few-public-methods,redefined-outer-name,use-dict-literal,iterating-modified-dict

```

Comment:
Are those necessary? I feel like `useless-suppression` could be avoided here?

Reply:
maybe the tests env is different than testing locally? when I run locally I get a message these are needed... e.g.

```
 Unexpected in testdata:
E          2: useless-suppression
```

Reply:
Hmm, it says line 2, so isn't that raising a warning _about_ the warnings Daniel doesn't think you need to suppress? I think this passes with just adding `iterating-modified-dict`.

Reply:
All functional test should run with `suppressed-message`, `locally-disabled` and `useless-suppression` enabled. I've recently updated the test runner so if this is not the case this is a bug and not working as intended.
As a result of this, these shouldn't need to be added to any individual test file.

Test runner and how it loads configuration can be found at `pylint/testutils/lint_module_test.py`.

Reply:
@jacobtylerwalls  - This is the messages raised when I run pytest without adding the `useless-suppression` to the disabled list..

This is how I run the test from terminal when I get this message:

`python tests/test_functional.py -k "test_functional[modified_iterating]"`

It seems in the file you sent me that the constructor which disables these messages run in `tests_functional`.
I debugged it and it seems like I get the error because I don't have the `pylintrc` file under `\testutils\testing_pylintrc`

Why there is no empty file by default in the repo? is it a bug? Anyways I would have solved it by adding the message disables to a `finally` block :)

code:

```
        rc_file: Union[Path, str] = PYLINTRC
        try:
            rc_file = test_file.option_file
            self._linter.disable("suppressed-message")
            self._linter.disable("locally-disabled")
            self._linter.disable("useless-suppression")
        except NoFileError:
            pass
```

Reply:
Ah, `testing_pylintrc` was a fairly recent addition. See https://github.com/PyCQA/pylint/commit/efe59ca44b9ed350c9790d79a39a7343af5e7ea7
You might have created your local branch when this commit was not yet on `main`. If you pull and merge `main` into your local branch this should be fixed!

Reply:
it happens when I install pylint as a package. This is when it doesn't create the file. maybe a bug?
(in my local repo I have it)

@DanielNoord 
Also, I think there is a bug in the code I attached, because even if the `.rc`  file doesn't exist we still want to have these disables...

I could upload a quick fix on another branch if you think it's a good idea

Reply:
Do you mean installing with `pip install pylint` or with `pip install -e .`? 

Hmm, really?
This is the relevant code:
https://github.com/PyCQA/pylint/blob/fdd71ee0cad6dc965c4a7bd7bc3d16e2ab564f70/pylint/testutils/lint_module_test.py#L47-L63
It sets `rc_file` to `testing_pylintrc` which includes the disables. Then it checks if there is a specific `.rc` file for the tests and overwrite `rc_file`. To make sure we still disable these three messages we then disable them manually on L50-52. I think this should disable all three in all cases, but I would obviously welcome a fix if I'm missing something!